ArduCopter/control_guided.cpp

```diff
@@ -212,6 +213,30 @@ void Copter::guided_set_angle(const Quaternion &q, float climb_rate_cms)
     guided_angle_state.update_time_ms = millis();
 }
 
+// guided_reached_destination_notify - notify if we've reached the destination
+// on guided mode - if we're taking off we consider altitude
+void Copter::guided_reached_destination_notify(bool check_altitude)
+{
+    bool reached_alt = true;
+
+    if (guided_prev_mode != guided_mode)
+        guided_reached_notified = false;
```

Comment:
@magicrub as far as I could see the loiter mode on plane and rover doesn't work as the guided mode on copters, the loiter start_time_ms is always reset when a new nav command is sent to APM, this doesn't happen to guided for copters.

The code you're changing in the mentioned commits use that condition to identify a mode has changed, since we got a new nav command the start_time_ms is set to zero and we can assume we're transitioning between modes or maybe we're "about to process a new command", so we're ok to send the item reached message based on that assumption, but we don't have the very same behavior for copters.

Am I maybe missing something here?

Thanks...